src/lib/core/say.c

```diff
@@ -1153,6 +1222,8 @@ write_to_syslog(struct log *log, int total)
 		if (log->fd >= 0)
 			close(log->fd);
 		log->fd = log_syslog_connect(log);
+		log->isatty = isatty(log->fd);
+		log->isatty_stderr = isatty(STDERR_FILENO);
```

Comment:
This chunk is not needed, since we have already set these fields in log_syslog_connect